sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -450,7 +450,8 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
         input => rawParser.parse(input, createParser, UTF8String.fromString),
         parsedOptions.parseMode,
         schema,
-        parsedOptions.columnNameOfCorruptRecord)
+        parsedOptions.columnNameOfCorruptRecord,
+        optimizeEmptySchema = true)
```

Comment:
No, no I'm just wondering since you made it a parameter that you can turn off and on, what would be the case to turn it off?

If there is none, shouldn't we just get rid of the parameter altogether ?

Reply:
> what would be the case to turn it off?

We can apply the optimization if we know in advance that one JSON object corresponds to one struct. In that case, we can return empty row if required schema (struct) is empty. If `multiLine` is enabled, there could be many structs per a JSON document. So, we cannot say in advance how many empty rows need to return without parsing.